### ISSUE TITLE
Attractor-model annotation conflates exact implicit differentiation with one-step and phantom gradients

### ISSUE THREAD

**Issue by microprediction** [microprediction/latentreasoning#154]
## Reviewed source

- Default-branch commit: [`88d1db6b1c014f641f67ec4b84ec1f51e8f0c43b`](https://github.com/microprediction/latentreasoning/commit/88d1db6b1c014f641f67ec4b84ec1f51e8f0c43b)
- Affected annotation: [`bibliography.html` lines 1796–1798](https://github.com/microprediction/latentreasoning/blob/88d1db6b1c014f641f67ec4b84ec1f51e8f0c43b/bibliography.html#L1796-L1798)
- Primary source: Fein-Ashley & Rashidinejad, [*Solve the Loop: Attractor Models for Language and Reasoning*, arXiv:2605.12466v1](https://arxiv.org/html/2605.12466)

The entry says that Attractor Models solve an output-embedding fixed point and “differentiate through it by the implicit function theorem.” The fixed-point part is correct. The gradient description conflates the exact implicit derivative with the cheaper surrogate estimators used in most reported regimes.

## Exact implicit differentiation versus the implemented estimators

For a fixed point `y* = T(y*, y0)` and output gradient `v = dL/dy*`, exact implicit differentiation requires solving

```
(I - J_y^T) u = v
```

and then using `u` in the parameter gradient. The paper derives this formula in §3.2 and Appendix B.1.

But [§3.2](https://arxiv.org/html/2605.12466#S3.SS2) then explicitly says:

> we use the one-step approximation u ≈ v

This drops the inverse-Jacobian factor entirely. It is Jacobian-free backpropagation (JFB), not the exact gradient supplied by the implicit function theorem. The paper's own [Table 5](https://arxiv.org/html/2605.12466#S4.T5) distinguishes all three estimators:

- full IFT: Anderson solve of the adjoint linear system;
- phantom gradient: a finite truncated approximation;
- one-step: `u ≈ v`.

The distinction is operationally large in that ablation. Relative to one-step, full IFT uses **4.8×** peak training memory and **2.7×** step time, for a 0.14 validation-perplexity improvement.

## The reported experiments use different backward regimes

The paper does not use one uniform “differentiate by IFT” recipe:

- The 140M and 370M language models use the one-step estimator, according to Table 8.
- The 770M language model lists an Anderson backward solve.
- The Sudoku/Maze reasoning experiments use a phantom-gradient scheme, not one-step or full IFT ([§4.2](https://arxiv.org/html/2605.12466#S4.SS2)).

Thus the site collapses three different gradient/resource regimes into the exact one. Calling all of them “implicit differentiation” also obscures why the neighboring bibliography entries separately define phantom gradients as deliberately inexact and JFB as dropping the Jacobian term.

## Consequence for the theoretical interpretation

This is not only terminology. Appendix B.3 argues that the exact implicit gradient contains `(I-J)^-1`, which becomes singular as an eigenvalue approaches 1 and therefore creates a barrier near loss of contraction. That mechanism is absent from the one-step update `u=v`: it contains no inverse and cannot diverge as `J` approaches the unit boundary.

The appendix later says that a one-step “truncated Neumann series” also diverges as the spectral radius approaches 1, but a one-step truncation is just the identity term. It remains finite. Only an infinite Neumann series can diverge at the boundary; a fixed finite truncation remains finite (though its norm can grow with the number of retained terms).

So the exact-IFT stability argument cannot simply be transferred to the one-step runs. JFB can still be a descent direction under separate conditions from Fung et al., and the empirical convergence results remain evidence, but those are different claims.

## Suggested correction

For example:

> Attractor Models refine a backbone proposal by root-finding an output-embedding fixed point, avoiding backpropagation through every forward solver step. Their experiments use several backward estimators: one-step Jacobian-free updates for the 140M/370M language models, an Anderson adjoint solve at 770M, and phantom gradients for the small reasoning models. Exact IFT is derived and ablated, but is not the uniform training rule; in the ablation it costs 4.8× the memory and 2.7× the step time of one-step JFB. Training memory therefore stays independent of forward solver depth, while estimator accuracy and constant factors differ by regime.

The fixed-point and constant-effective-depth-memory contributions can remain, but the entry should distinguish exact implicit differentiation from the surrogate gradients that make most experiments cheap.

**Comment by microprediction** [comment]
Confirmed and applied in ff85fa5. §3.2 says 'we use the one-step approximation u ≈ v'. Table 8 lists one-step IFT at 140M and 370M and Anderson at 770M, and §4.2 uses the phantom-gradient scheme for the reasoning tasks. Table 5 puts full IFT at 4.8× memory and 2.7× step time for 0.14 PPL. The entry now names each estimator.
